xds/internal/balancer/clusterresolver/e2e_test/eds_impl_test.go

```diff
@@ -1138,3 +1143,168 @@ func waitForProducedZeroAddressesError(ctx context.Context, t *testing.T, client
 	}
 	return errors.New("timeout when waiting for RPCs to fail with UNAVAILABLE status and produced zero addresses")
 }
+
+// Test runs a server which listens on multiple ports. The test updates xds resouce
+// cache to contain a single endpoint with multiple addresses. The test intercepts
+// the resolver updates sent to the petiole policy and verifies that the
+// additional endpoint addresses are correctly propagated.
+func (s) TestEDS_EndpointWithMultipleAddresses(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	// Start a backend server which provide that listens to multiple ports to simulate
+	// a backend with multiple addresses.
+	servers, cleanup2 := startTestServiceBackends(t, 1)
```

Comment:
Nit: should we simply start a stub server here instead of using `startTestServiceBackends` since we have a single server and we can avoid a bunch of `servers[0]`.

Also: s/cleanup2/cleanup/

Also super nit: A newline after the `defer cancel` and this block would be nice.